app/controllers/api/v1/city_info_controller.rb

```diff
@@ -0,0 +1,30 @@
+class Api::V1::CityInfoController < ApplicationController
+
+
+  def index
+    if valid_header?
+      render :json => JSON.parse(city_info_facade.return_info.to_json()), :status => :ok
+    else
+      invalid_header = {
+        error: "Missing location header",
+        status: 400
+      }
+        render :json => invalid_header, :status => :unauthorized
```

Comment:
Looks like you're sending back a `401` on this line, but a `400` in the JSON status on line 10? Is that right? Should it be a `406 Bad Request` instead?